flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/StreamTaskTimerTest.java

```diff
@@ -70,11 +70,22 @@ public void teardown() throws Exception {
     }
 
     @Test
-    public void testOpenCloseAndTimestamps() {
-        // first one spawns thread
+    public void testOpenCloseAndTimestamps() throws InterruptedException {
+        // Wait for the start of threads: throughput calculation.
+        int initiallyExpectedThreads = 1;
+        waitUntilTaskTriggerThreadsOnStart(initiallyExpectedThreads);
+
+        // Spawn one more thread.
         timeService.registerTimer(System.currentTimeMillis(), timestamp -> {});
```

Comment:
Maybe removing this line directly, and use the scheduled calculating throughput directly?

I am not sure scheudle a new timer will 100% to spawn a new thread.

Reply:
I rewrote the test